packages/flutter/lib/src/cupertino/list_tile.dart

```diff
@@ -0,0 +1,415 @@
+    if (widget.subtitle != null) {
+      subtitle = DefaultTextStyle(
+        style: subtitleTextStyle,
+        maxLines: 1,
+        child: widget.subtitle!,
+      );
+    }
+
+    Widget? additionalInfo;
+    if (widget.additionalInfo != null) {
+      additionalInfo = DefaultTextStyle(
+        style: additionalInfoTextStyle!,
+        maxLines: 1,
+        child: widget.additionalInfo!,
+      );
+    }
+
+    // The color for default state tile is set to either what user provided or
+    // null and it will resolve to the correct color provided by context. But if
+    // the tile was tapped, it is set to what user provided or if null to the
+    // default color that matched the iOS-style.
+    Color? backgroundColor = widget.backgroundColor;
+    if (_tapped) {
+      backgroundColor = widget.backgroundColorActivated ?? CupertinoColors.systemGrey4.resolveFrom(context);
+    }
+
+    double minHeight;
+    switch (widget._type) {
+      case _CupertinoListTileType.base:
+        minHeight = subtitle == null ? _kMinHeight : _kMinHeightWithSubtitle;
+        break;
+      case _CupertinoListTileType.notched:
+        minHeight = widget.leading == null ? _kNotchedMinHeightWithoutLeading : _kNotchedMinHeight;
+        break;
+    }
+
+    final Widget child = Container(
+      constraints: BoxConstraints(minWidth: double.infinity, minHeight: minHeight),
+      color: backgroundColor,
+      child: Padding(
+        padding: padding,
+        child: Row(
+          children: <Widget>[
+            if (widget.leading != null) ...<Widget>[
+              SizedBox(
+                width: widget.leadingSize,
+                height: widget.leadingSize,
+                child: Center(
+                  child: widget.leading,
+                ),
+              ),
+              SizedBox(width: widget.leadingToTitle),
+            ] else
+              SizedBox(height: widget.leadingSize),
+            Column(
+              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+              crossAxisAlignment: CrossAxisAlignment.start,
+              children: <Widget>[
+                title,
+                if (subtitle != null) ...<Widget>[
+                  const SizedBox(height: _kNotchedTitleToSubtitle),
+                  subtitle,
+                ],
+              ],
+            ),
+            const Spacer(),
+            if (additionalInfo != null) ...<Widget>[
+              additionalInfo,
+              if (widget.trailing != null)
+                const SizedBox(width: _kAdditionalInfoToTrailing),
+            ],
+            if (widget.trailing != null) widget.trailing!
+          ],
+        ),
+      ),
+    );
+
+    if (widget.onTap == null)
+      return child;
+
+    return GestureDetector(
+      onTapDown: (_) => setState(() => _tapped = true),
+      onTapCancel: () => setState(() => _tapped = false),
+      onTap: () async {
+        await widget.onTap!();
+        setState(() => _tapped = false);
```

Comment:
this is a trivial nit but generally we prefer `setState(() { ... })` since it doesn't actually return anything

Reply:
Will change all such occurrences.